src/mscorlib/src/System/Diagnostics/Eventing/XplatEventLogger.cs

```diff
@@ -109,24 +109,51 @@ private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnly
                 sb.Append("\\\"");
                 sb.Append(':');
 
-                var valuestr = payload[i] as string;
-
-                if( valuestr != null)
-                {
-                    sb.Append("\\\"");
-                    minimalJsonserializer(valuestr,sb);
-                    sb.Append("\\\"");
-                }
-                else
+                switch(payload[i])
                 {
-                    sb.Append(payload[i].ToString());
+                    case string str:
+                    {
+                        sb.Append("\\\"");
+                        minimalJsonserializer(str, sb);
+                        sb.Append("\\\"");
+                        break;
+                    }
+                    case byte[] byteArr:
+                    {
+                        sb.Append("\\\"");
+                        AppendByteArrayAsHexString(sb, byteArr);
+                        sb.Append("\\\"");
+                        break;
+                    }
+                    default:
+                    {
+                        if(payload[i] != null)
+                        {
+                            sb.Append(payload[i].ToString());
+                        }
+                        break;
+                    }
                 }
-
             }
             sb.Append('}');
             return StringBuilderCache.GetStringAndRelease(sb);
         }
 
+        private static void AppendByteArrayAsHexString(StringBuilder builder, byte[] byteArray)
+        {
+            Debug.Assert(builder != null);
+            Debug.Assert(byteArray != null);
+
+            ReadOnlySpan<char> hexFormat = "X2";
+            Span<char> hex = stackalloc char[2];
```

Comment:
Will all characters always be ASCII?

Reply:
The resulting string will always contain ASCII because this just converts a blob to hex.